### PR TITLE
Removes HelixPropertyStoreFactory

### PR DESCRIPTION
**Pull request by xiahome**
Instead, helixPropertyStore will be instantiated in both
getHelixNotifier() and getHelixStoreOperator() call.

Test:
./gradlew clean && ./gradlew build

### CODE REVIEW

File: ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java [hunk 1]
```diff
@@ -49,23 +54,25 @@
...
   public void setup() throws Exception {
+    storeKeyToMockStoreMap.clear();
+    helixNotifier = new HelixNotifier(getMockHelixStore(storeConfig));
+    helixNotifier2 = new HelixNotifier(getMockHelixStore(storeConfig));
     resetReferenceTopicsAndMessages();
     resetListeners();
-    storeFactory = new MockHelixPropertyStoreFactory<>(false, false);
-    HelixPropertyStoreUtils.deleteStoreIfExists(storeConfig, storeFactory);
-    notifier = new HelixNotifier(storeConfig, storeFactory);
+    deleteStoreIfExists();
```
Comment (vgkholla) [inline ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java]: could you move this to the top?

File: ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java [hunk 2]
```diff
@@ -49,23 +54,25 @@
...
+          STORAGE_ROOT_PATH);
+  private final Map<String, MockHelixPropertyStore<ZNRecord>> storeKeyToMockStoreMap = new HashMap<>();
+  private HelixNotifier helixNotifier;
+  private HelixNotifier helixNotifier2;
 
   @Before
   public void setup() throws Exception {
+    storeKeyToMockStoreMap.clear();
+    helixNotifier = new HelixNotifier(getMockHelixStore(storeConfig));
+    helixNotifier2 = new HelixNotifier(getMockHelixStore(storeConfig));
```
Comment (vgkholla) [inline ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java]: can we have this only in the test that it is required in?

File: ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java [hunk 3]
```diff
@@ -42,23 +46,21 @@
 public class HelixNotifier implements Notifier<String> {
   private static final String TOPIC_PATH = "/topics";
   private static final String MESSAGE_KEY = "message";
+  private static final Logger classLogger = LoggerFactory.getLogger(HelixNotifier.class);
```
Comment (xiahome) [inline ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java]: Would follow a naming convention that `ALL_CAPS` are used for `static final` primitives, and `camelCase` for non-primitives. [Link](https://stackoverflow.com/questions/1417190/should-a-static-final-logger-be-declared-in-upper-case)

File: ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java [hunk 4]
```diff
@@ -142,6 +144,32 @@ public void unsubscribe(String topic, TopicListener<String> topicListener) {
   }
 
   /**
+   * A helper method that gets a {@link HelixNotifier}. Each call of this method will generate a different
+   * {@link HelixNotifier} instance.
+   * @param storeConfig A {@link HelixPropertyStore} used to instantiate a {@link HelixNotifier}.
+   * @return A {@link HelixNotifier}.
+   */
+  public static HelixNotifier getHelixNotifier(HelixPropertyStoreConfig storeConfig) {
```
Comment (nsivabalan) [inline ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java]: who are the callers of this method? I feel that this should be one of the constructor.

File: ambry-commons/src/test/java/com.github.ambry.commons/HelixStoreOperator.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.utils.TestUtils.*;
+
+
+/**
+ * An agent to perform operations on a {@link HelixPropertyStore}.
+ */
+class HelixStoreOperator {
```
Comment (nsivabalan) [inline ambry-commons/src/test/java/com.github.ambry.commons/HelixStoreOperator.java]: is there any change in this class apart from moving from a diff file

Comment (xiahome) [inline ambry-commons/src/test/java/com.github.ambry.commons/HelixStoreOperator.java]: No it was just some copy & paste

File: ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java (hunk 4 above)
Comment (xiahome) [inline ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java]: This method serves as a constructor for `HelixNotifier`. The reason not to put it into a constructor depends on if we want to control the helix property store the notifier depends on. I see values of doing on either side, but have some decoupling is not a bad thing, as we can control what to pass in. However, the caller may not want to get involved into the complication of instantiating a propertystore, so we introduce this helper method.

File: ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java (hunk 2 above)
Comment (xiahome) [inline ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java]: Fixed.

File: ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java (hunk 1 above)
Comment (xiahome) [inline ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java]: Moved up.

File: ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java (hunk 4 above)
Comment (xiahome) [inline ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java]: Have made it as a constructor after offline discussion.

**Comment by xiahome** [comment]
@vgkholla @nsivabalan Addressed your comments.

**Comment by nsivabalan** [comment]
Looks good. Merging.

**Pull request merged**